src/middlewared/middlewared/plugins/service_monitor.py

```diff
@@ -137,40 +147,58 @@ def run(self):
             enabled = self.isEnabled(self.name)
 
             self.logger.debug("[ServiceMonitorThread] connected=%s started=%s enabled=%s", connected, started, enabled)
+            # Everithing is OK
+            if connected and started and enabled:
+                # Do we want to reset all alerts when things get back to normal?
+                ServiceMonitorThread.reset_alerts(self.name)
+                ntries = 0
+                continue
 
-            if (connected is False):
-                self.alert("attempt %d to recover service %s\n" % (ntries + 1, self.name))
+            start_service = False
+            stop_service = False
+            ntries += 1
 
-            if (connected is True) and (started is False):
-                self.logger.debug("[ServiceMonitorThread] enabling service %s", self.name)
+            self.alert("attempt %d to recover service %s\n" % (ntries, self.name))
+
+            if connected:
+                if not started:
+                    start_service = True
+            else:
+                if enabled:
+                    stop_service = True
+
+            if stop_service:
+                self.logger.debug("[ServiceMonitorThread] disabling service %s", self.name)
                 try:
-                    self.middleware.call_sync('service.start', self.name)
+                    self.middleware.call_sync('service.stop', self.name)
                 except Exception:
```

Comment:
Exception should not happen here, `service.start` would just return False. We should at least log the traceback.